src/routes/chat/+page.svelte

```diff
@@ -645,7 +650,14 @@
         {/if}
 
         {#if $activeChat.users.length > 0}
-            <Chatbar filesSelected={files} replyTo={replyTo} on:onsend={_ => (files = [])}>
+            <Chatbar
+                filesSelected={files}
+                replyTo={replyTo}
+                typing={$activeChat.typing_indicator.users && $activeChat.typing_indicator.users().map(u => $users[u])}
+                on:onsend={_ => (files = [])}
+                on:input={_ => {
+                    typing()
```

Comment:
Do we really want to be sending the event on every input? It might make sense to send it only once within a specific timeframe and resend if one is still typing and on the other side to have a timeout to stop the indicator after the timer has fired off or if a cancel event is received.

Reply:
We can debounce this with a simple debounce function.

`function debounce<T extends (...args: any[]) => void>(func: T, wait: number): (...args: Parameters<T>) => void {
    let timeoutId: ReturnType<typeof setTimeout> | undefined;

    return function(...args: Parameters<T>): void {
        if (timeoutId) {
            clearTimeout(timeoutId);
        }

        timeoutId = setTimeout(() => {
            func(...args);
        }, wait);
    };
}

export default debounce;
`

Reply:
This function could be added to the utils.ts file